### ISSUE TITLE
GGMAX Bug - Fog Settings in Editor

### ISSUE THREAD

**Issue by JCLEONITA** [TheGameCreators/GameGuruRepo#5798]
want to be able to remove the fog every time  enter a building, and have it return when  leave the building to simulate going indoors and fog go away and exit outdoor and fog come in again.

Plus there is a bug because the fog has to be off when you start and triggered to get foggy before you can turn it off with a switch or zone because if you have it foggy first the behavior wont work ..

here a video

https://discord.com/channels/950796460683829329/950796461136826389/1300160944231415830

https://streamable.com/on5fx2

**Comment by Necrym59** [comment]
No the fog settings at the time of game start is read in by the behavior when you activate to increase the fog it is changed and if activated a second time will receed it back to the start settings.
Best way is to use two zones to do this.

**Comment by JCLEONITA** [comment]
I think the Problem is .. If i set the fog slider now in the editor for my level i expect it to stay that way when i run testgame. 
The behaviour is actually moving the slider in testgame and when you exit back to the editor and the slider has set moved to the setting you left the game whether fog close or fog far ..

**Comment by JCLEONITA** [comment]
> No the fog settings at the time of game start is read in by the behavior when you activate to increase the fog it is changed and if activated a second time will receed it back to the start settings. Best way is to use two zones to do this to achieve what you want.

but hte behavior dont act like this because if i set a value in the editor for e.g 2 -10 and after i activate gthe behavior when player is  indoors for e.g. 30 -50 , the global fog value is set to a value that is none of them

**Comment by JCLEONITA** [comment]
and eventually how can i use two zones in the same place since player is intended to enter and exit the building several times as  he likes, this way zone will overlap eachother

**Comment by JCLEONITA** [comment]
> No the fog settings at the time of game start is read in by the behavior when you activate to increase the fog it is changed and if activated a second time will receed it back to the start settings.

i want the behavior to descrease the fog when activated a first time and receed to the start settings when activated a second time ( exit the building)

**Comment by Necrym59** [comment]
The fog setting are changed when you enter a multi-use zone the first time and reset back to the defaults the second time the zone is entered. If by chance you are in test game and exit after you have enetered the zone once then yes max is keeping the updated settings and should reset them to the original default settings set in the editor, i hjave no control over this.

**Comment by JCLEONITA** [comment]
> f by chance you are in test game and exit after you have entered the zone once then yes max is keeping the updated settings and should reset them to the original default settings set in the editor, i have no control over this.

exaclty this is a bug i suppose since in editor values should  reset to the original player  chosen values

**Comment by Kasseyus** [comment]
Standard Fog:
![image](https://github.com/user-attachments/assets/589b7b1d-11ef-476f-9278-c4db7434cdad)

Fog Behaviour:
![image](https://github.com/user-attachments/assets/7b384548-09d8-493e-8869-d19c3d7a3c23)

Enter Trigger Zone and Wait:
![image](https://github.com/user-attachments/assets/c428e8cf-8f9b-4d24-9ac1-03c03d23ef9b)

MAX Fog Settings Gone Wild:
![image](https://github.com/user-attachments/assets/7abcd7a0-7c02-4b61-8cb8-77700bb2cf8c)

**Comment by Kasseyus** [comment]
@LeeBamberTGC Fog Script seems to be not working as intended.

**Comment by JCLEONITA** [comment]
fog scripts a a lot of issue ..some i reported ,many other i didnt since i think i really broken.. and it s really strange that lots of peoples using max in these years didnt notice them... 

sometimes when i change fog sentting in test mode by zone or switch and return to  editor the editor is all white screen i need to go in test mode again 2/3 times and after i return to the editor it shows correctly--- many times not... and many more issues i noticed here and there using it

**Comment by Necrym59** [comment]
The fog script works. The script isnt the problem, the problem is the editor isnt resetting back to its start setings when leaving test play. Do get your facts straight and description of the problem correct or we cant help.

**Comment by JCLEONITA** [comment]
> The fog script works. The script isnt the problem, the problem is the editor isnt resetting back to its start setings when leaving test play. Do get your facts straight and description of the problem correct or we cant help.

yes the fog per se works correctly as you said but there are some engine issue with it that maybe nneds to be addressed i suppose

firstly the fact that as you already stated the editor doenst reset the value when exit from test game .. and maybe this can be related to my other issue of the blank screen  after test game exit..

**Comment by Kasseyus** [comment]
> ...and maybe this can be related to my other issue of the blank screen after test game exit..

@JCLEONITA This sounds like a separate issue.  Can you please raise a new ticket, with as much detail as possible about this and I will be able to investigate further.

**Comment by LeeBamberTGC** [comment]
@JCLEONITA Check with Necrym59 for script specific issues, the good bug report related to the LUA messing up the editor based fog settings has been fixed for the next DEVTEST build.
